### ISSUE TITLE
AKS can't provision azure_active_directory with managed = true, and crashes after this has been configured manually

### ISSUE THREAD

**Issue by gercograndia** [terraform-providers/terraform-provider-azurerm#8782]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.13.2
+ provider registry.terraform.io/hashicorp/azuread v0.7.0
+ provider registry.terraform.io/hashicorp/azurerm v2.17.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "cluster" {
  name                      = "cluster-${local.full_name}"
  location                  = var.region
  resource_group_name       = local.resource_group.name
  dns_prefix                = local.full_name
  private_cluster_enabled   = true
  kubernetes_version        = var.default_node_pool.kubernetes_version
  node_resource_group       = "${local.resource_group.name}-nodes"

  addon_profile {
    aci_connector_linux {
      enabled = false
    }
    kube_dashboard {
      enabled = true
    }
    oms_agent {
      enabled = true
      log_analytics_workspace_id = azurerm_log_analytics_workspace.monitoring.id
    }
    http_application_routing {
      enabled = true
    }
  }

  network_profile {
    network_plugin        = "kubenet"
    load_balancer_sku     = "Standard"
    outbound_type         = "loadBalancer"
  }

  default_node_pool {
    name                  = var.default_node_pool.name
    vnet_subnet_id        = azurerm_subnet.k8s.id
    node_count            = var.default_node_pool.node_count
    vm_size               = var.default_node_pool.node_type
    enable_node_public_ip = false # public cluster are not supported, security
    os_disk_size_gb       = var.default_node_pool.os_disk_size_gb
    tags                  = local.tags 
  }

  identity {
    type = "SystemAssigned"
  }
  
  role_based_access_control {
    enabled               = var.enable_rbac

    azure_active_directory {
      managed = 
      admin_group_object_ids = ["5bxxxxxxxxxxxxxxxxxb47"]
    }
  }

  tags = local.tags

  timeouts {
    create = "30m"
    delete = "30m"
  }
}
```

### Debug Output

When trying to update the (already existing cluster) with `azure_active_directory` information, it fails with the following error:

Error: updating Managed Kubernetes Cluster AAD Profile in cluster "cluster-aks-demo-full-tst" (Resource Group "rg-aks-demo-full-tst"): containerservice.ManagedClustersClient#ResetAADProfile: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The value of parameter aadProfile.clientAppID is invalid. Please see https://aka.ms/aks-naming-rules for more details." Target="aadProfile.clientAppID"

However, updating the cluster via a `az aks update .... --enable-aad` command works fine.

### Panic Output

The panic output happens AFTER the existing cluster has been updated to use enable-aad, and when the hcl does NOT contain the `azure-active-directory` information. Hence it tries to remove that configuration item and then it crashes heavily. Possibly because you can't remove it, as this seems also not possible via the az cli and portal.

https://gist.github.com/gercograndia/b080a2e9469e13ec9942ee1475d0b945

### Expected Behavior

Enabling the `azure_active_directory` on the cluster should be possible given the config as listed above. This is also my understanding of the documentation, and works sort of in a same way via the cli or portal.

And when reverting back is not possible, terraform should not panic but just report this is not supported.

### Actual Behavior

Already described above.

### Steps to Reproduce

1. Create AKS cluster with no `azure_active_directory` block.
2. Then add this block with `managed = true` and optionally one or more `admin_group_object_ids`. 
3. Apply, the `The value of parameter aadProfile.clientAppID is invalid` will be raised.
4. Update the aks cluster via the cli with `--enable-aad` and optionally one or more `admin_group_object_ids`.
5. Run again an apply with terraform with NO `azure_active_directory` block. 
6. Terraform will error and most of the time panic.

### Important Factoids

Nope

### References

* #0000

**Comment by mozts2005** [comment]
You need to use version 2.21.0 or above. see [PR #7874](https://github.com/terraform-providers/terraform-provider-azurerm/pull/7874)

**Comment by gercograndia** [comment]
Ah, thanks, I'll give that a go.

**Comment by gercograndia** [comment]
That indeed resolved the issue, running v2.21.0.

I'll close the issue.
